orangecontrib/recommendation/widgets/owbrismf.py

```diff
@@ -125,6 +153,10 @@ def add_main_layout(self):
                                  callback=self.settings_changed)
         self.settings_changed()  # Update (extra) settings
 
+    def setup_layout(self):
+        super().setup_layout()
+        gui.button(self.apply_button, self, "Cancel", callback=self.cancel)
```

Comment:
This button should probably be disabled when a task is done.
Besides, after the task is interrupted, could it be re-run? If yes, button's text (and it's functionality) should be replaced with `Resume` or something similar.